components/datadog/apps/nginx/k8s/modifiers.go

```
@@ -0,0 +1,164 @@
+	// buffer
+	merged := pulumi.StringMap{}
+	// get everything from left map
+	for k, v := range left {
+		merged[k] = v
+	}
+	// get everything from right map, this will overwrite if
+	// keys are duplicated
+	for k, v := range right {
+		merged[k] = v
+	}
+
+	return merged
+}
+
+// ensureDeploymentPodTemplateSpec performs nil and type checks
+// on a DeploymentArgs struct and returns PodSpecArgs
+func ensureDeploymentPodTemplateSpec(d *appsv1.DeploymentArgs) (*corev1.PodSpecArgs, bool) {
+	// nil check spec
+	deploymentSpecPtr := d.Spec
+	if deploymentSpecPtr == nil {
+		d.Spec = &appsv1.DeploymentSpecArgs{}
+	}
+	// type check spec
+	spec, ok := deploymentSpecPtr.(*appsv1.DeploymentSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check spec.Template
+	podTemplatePtr := spec.Template
+	if podTemplatePtr == nil {
+		spec.Template = &corev1.PodTemplateSpecArgs{}
+	}
+	// type check spec.Template
+	podTemplate, ok := podTemplatePtr.(*corev1.PodTemplateSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check spec.Template.Spec
+	podTemplateSpecPtr := podTemplate.Spec
+	if podTemplateSpecPtr == nil {
+		podTemplate.Spec = &corev1.PodSpecArgs{}
+	}
+
+	// type check spec.Template.Spec
+	podTemplateSpec, ok := podTemplateSpecPtr.(*corev1.PodSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	return podTemplateSpec, true
+}
+
+// WithRuntimeClass sets a deployment's RuntimeClassName
+func WithRuntimeClass(rtc string) DeploymentModifier {
+	return func(d *appsv1.DeploymentArgs) {
+		if podTemplateSpec, ok := ensureDeploymentPodTemplateSpec(d); ok {
+			podTemplateSpec.RuntimeClassName = runtimeClassToPulumi(rtc)
+		}
+	}
+}
+
+// WithServiceAccount sets a deployment's ServiceAccount
+func WithServiceAccount(serviceAccount *corev1.ServiceAccount) DeploymentModifier {
+	return func(d *appsv1.DeploymentArgs) {
+		if podTemplateSpec, ok := ensureDeploymentPodTemplateSpec(d); ok {
+			podTemplateSpec.ServiceAccount = serviceAccount.Metadata.Name()
+		}
```

Errors (when `ok` is `false`) are silently ignored.
Should we raise an error?